src/lib/y2storage/proposal/autoinst_nfs_planner.rb

```diff
@@ -0,0 +1,212 @@
+      #
+      # @return [Boolean]
+      def valid_drive?(drive, partition: nil, format: :new)
+        partition_section = partition || drive.partitions.first
+
+        !missing_drive_values?(drive, format) &&
+          !missing_partition_values?(partition_section, format)
+      end
+
+      # Whether any value is missing for the drive section
+      #
+      # Errors are registered when values are missing.
+      #
+      # @param drive [AutoinstProfile::DriveSection]
+      # @param format [:new, :old] whether the section is using the new or old AutoYaST style
+      #
+      # @return [Boolean]
+      def missing_drive_values?(drive, format)
+        missing_any_value?(drive, mandatory_drive_values(format))
+      end
+
+      # Whether any value is missing for the partition section
+      #
+      # Errors are registered when values are missing.
+      #
+      # @param partition_section [AutoinstProfile::PartitionSection]
+      # @param format [:new, :old] whether the section is using the new or old AutoYaST style
+      #
+      # @return [Boolean]
+      def missing_partition_values?(partition_section, format)
+        missing_any_value?(partition_section, mandatory_partition_values(format))
+      end
+
+      # Whether any of the given values is missing in the given section
+      #
+      # Note: finding the first missing value is faster, but all values are checked to
+      # register all possible issues.
+      #
+      # @param section [AutoinstProfile::SectionWithAttributes]
+      # @param values [Array<Symbol>]
+      #
+      # @return [Boolean]
+      def missing_any_value?(section, values)
+        values.map { |v| missing_value?(section, v) }.any?
+      end
+
+      # Whether the given value is missing in the given section
+      #
+      # An error is registered when the value is missing.
+      #
+      # @param section [AutoinstProfile::SectionWithAttributes]
+      # @param value [Symbol]
+      #
+      # @return [Boolean]
+      def missing_value?(section, value)
+        return false if section.send(value)
+
+        issues_list.add(:missing_value, section, value)
+
+        true
+      end
+
+      # Mandatory values for the drive section
+      #
+      # @param format [:new, :old] whether the section is using the new or old AutoYaST style
+      # @return [Array<Symbol>]
+      def mandatory_drive_values(format)
+        mandatory_values(format)[:drive]
+      end
+
+      # Mandatory values for the partition section
+      #
+      # @param format [:new, :old] whether the section is using the new or old AutoYaST style
+      # @return [Array<Symbol>]
+      def mandatory_partition_values(format)
+        mandatory_values(format)[:partition]
+      end
+
+      # Mandatory values depending on the AutoYaST style
+      #
+      # @param format [:new, :old] new or old AutoYaST style
+      # @return [Hash<Symbol, Array<Symbol>>]
```

Comment:
here link documentation for constant or document it here and link constants to this method. But knowing meaning of symbols in hash is important to be written.